pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp

```diff
@@ -0,0 +1,1837 @@
+      return op.emitError("Cannot process non-unit pooling dilation.");
+
+    // FIXME: add ceil_mode support.
+
+    int64_t outputHDim =
+        ConvertAtenPoolingBaseOp<AtenOpT, MhloOpT>::getOutputDim(
+            inputShape[inputRank - 2], kernelSize[0], strideArray[0],
+            padArray[0], padArray[0], dilationArray[0]);
+    int64_t outputWDim =
+        ConvertAtenPoolingBaseOp<AtenOpT, MhloOpT>::getOutputDim(
+            inputShape[inputRank - 1], kernelSize[1], strideArray[1],
+            padArray[1], padArray[1], dilationArray[1]);
+    SmallVector<int64_t> outputShape;
+    if (inputRank > 3)
+      outputShape.push_back(inputShape[0]);
+    outputShape.push_back(outputHDim);
+    outputShape.push_back(outputWDim);
+    outputShape.push_back(inputShape[inputRank - 3]);
+    outputTy = RankedTensorType::get(outputShape, inputElemTy);
+
+    return success();
+  }
+};
+
+// Ref: Error checking based on the Torch to LinAlg lowering
+template <typename AtenOpT, int fillVal>
+class ConvertAtenConstPatternOp : public OpConversionPattern<AtenOpT> {
+public:
+  using OpConversionPattern<AtenOpT>::OpConversionPattern;
+  using OpAdaptor = typename AtenOpT::Adaptor;
+  LogicalResult
+  matchAndRewrite(AtenOpT op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    auto outType = OpConversionPattern<AtenOpT>::getTypeConverter()
+                       ->convertType(op.getType())
+                       .template dyn_cast<TensorType>();
+
+    if (!outType)
+      return op.emitError("Only Tensor types supported in MHLO");
+
+    Type outElemTy = outType.getElementType();
+    if (!outElemTy.isIntOrFloat())
+      return op.emitError(
+          "Only floating-point or integer datatype legalization supported");
+
+    // FIXME: Handle layout, device and pin_memory. Assume dtype has been
+    // processed to set output type correctly?
+    if (!op.layout().getType().template isa<Torch::NoneType>())
+      return op.emitError("Only default layout is supported");
+
+    bool pinMemory;
+    if (!op.pin_memory().getType().template isa<Torch::NoneType>() &&
+        (!matchPattern(op.pin_memory(), m_TorchConstantBool(&pinMemory)) ||
+         pinMemory)) {
+      return op.emitError(
+          "Unsupported pin_memory, should be either None or false");
+    }
+
+    SmallVector<int64_t> shape;
+    if (!matchPattern(op.size(), m_TorchConstantIntList(shape))) {
+      return op.emitError("Shape must be a list of Scalar constants");
+    }
+
+    int64_t size = 1;
+    for (auto s : shape)
+      size *= s;
+
+    SmallVector<int32_t> values(size, fillVal);
+    auto constOp =
+        mhlo::getConstTensor<int32_t>(rewriter, op, values, shape).getValue();
+
+    rewriter.replaceOpWithNewOp<mhlo::ConvertOp>(op, outType, constOp);
+
+    return success();
+  }
+};
+
+template <typename AtenOpT>
+class ConvertAtenFillScalarOp : public OpConversionPattern<AtenOpT> {
```

Comment:
Better to add FileCheck based UT for each conversion pattern

Reply:
I will add FileCheck UT in another PR: https://github.com/alibaba/BladeDISC/issues/366. I want to merge the PR quickly